### PR TITLE
Add rdma-core recipe

### PR DESCRIPTION
**Pull request by pentschev**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

<hr>

Fixes https://github.com/conda-forge/staged-recipes/issues/22306

### CODE REVIEW

File: recipes/rdma-core/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - {{ cdt("libnl3-devel") }}
+    - {{ cdt("systemd-devel") }}
+    - cmake
+    - make
```
Comment (kkraus14) [inline recipes/rdma-core/meta.yaml:25]: If we're using Ninja is there a reason we need `make` as well?

**Review by kkraus14 — APPROVED** [review]
One minor question in the recipe, otherwise LGTM

File: recipes/rdma-core/meta.yaml (hunk 1 above)
Comment (pentschev) [reply recipes/rdma-core/meta.yaml]: No reason, that was an oversight, thanks for catching. As of https://github.com/conda-forge/staged-recipes/pull/22869/commits/a9125a8ff5abf9f1b87aab9dd3da776c95c80001 that is now removed.

**Review by kkraus14 — APPROVED** [review]

**Comment by pentschev** [comment]
Thanks Keith for reviewing and merging this! 😄

**Pull request merged**